patches/cip-core/v2-0001-Add-option-to-use-swupdate-handler-roundrobin.patch

```diff
@@ -0,0 +1,182 @@
+Add the Option `SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO` to
+use the handler directly from the repository.
+
+The handler currently doesn't support secureboot.
+
+[1]:https://gitlab.com/cip-playground/swupdate-handler-roundrobin/
+
+Signed-off-by: Quirin Gylstorff <quirin.gylstorff@siemens.com>
+---
+ classes/swupdate-config.bbclass               | 12 ++++++---
+ kas/opt/ebg-secure-boot-base.yml              |  3 +++
+ .../files/secure-boot/sw-description.tmpl     |  2 +-
+ recipes-core/images/files/sw-description.tmpl | 21 ++++++++++-----
+ .../files/swupdate.handler.efibootguard.ini   | 26 +++++++++++++++++++
+ recipes-core/swupdate/swupdate.bb             |  9 ++++++-
+ 6 files changed, 62 insertions(+), 11 deletions(-)
+ create mode 100644 recipes-core/swupdate/files/swupdate.handler.efibootguard.ini
+
+diff --git a/classes/swupdate-config.bbclass b/classes/swupdate-config.bbclass
+index 4e46b37..a7a7aa4 100644
+--- a/classes/swupdate-config.bbclass
++++ b/classes/swupdate-config.bbclass
+@@ -21,9 +21,17 @@ KFEATURE_lua = ""
+ KFEATURE_lua[BUILD_DEB_DEPENDS] = "liblua5.3-dev"
+ KFEATURE_lua[KCONFIG_SNIPPETS] = "file://swupdate_defconfig_lua.snippet"
+ 
++SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO ?= "1"
++
++SRC_URI_append = " ${@ 'git://gitlab.com/cip-playground/swupdate-handler-roundrobin.git;protocol=https;destsuffix=swupdate-handler-roundrobin;name=swupdate-handler-roundrobin' \
++    if d.getVar('SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO') == '1' else '' \
++    }"
++SRCREV_swupdate-handler-roundrobin ?= "6ac9e49eaa4e866a3eda12eee3a223820ba8e0bf"
++SWUPDATE_LUASCRIPT ?= "swupdate-handler-roundrobin/swupdate_handlers_roundrobin.lua"
+ KFEATURE_luahandler = ""
+ KFEATURE_luahandler[KCONFIG_SNIPPETS] = "file://swupdate_defconfig_luahandler.snippet"
+-KFEATURE_luahandler[SRC_URI] = "file://${SWUPDATE_LUASCRIPT}"
++KFEATURE_luahandler[SRC_URI] = "${@ 'file://${SWUPDATE_LUASCRIPT}' \
++                                  if d.getVar('SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO') == '0' else '' }"
+ 
+ KFEATURE_DEPS = ""
+ KFEATURE_DEPS[luahandler] = "lua"
+@@ -57,8 +65,6 @@ KFEATURE_u-boot[DEBIAN_DEPENDS] = "${@ 'libubootenv0.1, u-boot-${MACHINE}-config
+ KFEATURE_u-boot[DEPENDS] = "libubootenv"
+ KFEATURE_u-boot[KCONFIG_SNIPPETS] = "file://swupdate_defconfig_u-boot.snippet"
+ 
+-SWUPDATE_LUASCRIPT ?= "swupdate_handlers.lua"
+-
+ def get_bootloader_featureset(d):
+     bootloader = d.getVar("SWUPDATE_BOOTLOADER", True) or ""
+     if bootloader == "efibootguard":
+diff --git a/kas/opt/ebg-secure-boot-base.yml b/kas/opt/ebg-secure-boot-base.yml
+index 30ca35a..484d0e5 100644
+--- a/kas/opt/ebg-secure-boot-base.yml
++++ b/kas/opt/ebg-secure-boot-base.yml
+@@ -16,3 +16,6 @@ local_conf_header:
+   initramfs: |
+     IMAGE_INSTALL += "initramfs-abrootfs-secureboot"
+     SWU_DESCRIPTION = "secureboot"
++  swupdate-secureboot: |
++    SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO = "0"
++    SWUPDATE_LUASCRIPT = "swupdate_handler.efibootguard.secureboot.lua"
```

Comment:
Where is this file `swupdate_handler.efibootguard.secureboot.lua`? And it should be one for U-Boot, not for EFI Boot Guard, right?